docs/specification/auth/credential.md

```diff
@@ -0,0 +1,119 @@
+---
+title: Credential Authentication
+type: docs
+weight: 52
+---
+
+{{< callout type="info" >}}
+**Protocol Revision**: {{< param protocolRevision >}}
+{{< /callout >}}
+
+The Model Context Protocol (MCP) supports credential-based authentication for scenarios requiring API keys or similar secrets.
+
+## Capabilities
+
+Clients supporting credential authentication **MUST** declare it during initialization:
+
+```json
+{
+  "capabilities": {
+    "auth": {
+      "credential": true
+    }
+  }
+}
+```
+
+Servers supporting credentials **MUST** include their capabilities:
+
+```json
+{
+  "capabilities": {
+    "auth": {
+      "credential": {
+        "list": true
+      }
+    }
+  }
+}
+```
+
+## Protocol Messages
+
+### Credential Requirements
+
+Servers can list required credentials using the credential/list capability:
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "result": {
+    "credential": [
+      {
+        "name": "X-API-KEY",
+        "description": "An API key must be provided to call this tool."
+      },
+      {
+        "name": "X-MISC-PASSWORD",
+        "description": "A password must be provided to list this resource"
+      }
+    ]
+  }
+}
+```
+
+### Providing Credentials
+
+Clients provide credentials through headers during initialization:
```

Comment:
`headers` is a bit confusing for non-HTTP transports, IMO. I think this should go into a `auth.credentials` field (or whatever we pick as the new name).